verify.go

```diff
@@ -171,12 +171,21 @@ func Verify(ctx context.Context, policyEnvelope dsse.Envelope, policyVerifiers [
 		dsse.VerifyWithIntermediates(intermediates...),
 		dsse.VerifyWithTimestampVerifiers(timestampVerifiers...),
 	)
-	accepted, err := pol.Verify(ctx, policy.WithSubjectDigests(vo.subjectDigests), policy.WithVerifiedSource(verifiedSource))
+
+	results, err := pol.Verify(ctx, policy.WithSubjectDigests(vo.subjectDigests), policy.WithVerifiedSource(verifiedSource))
 	if err != nil {
-		return nil, fmt.Errorf("failed to verify policy: %w", err)
+		return false, nil, fmt.Errorf("error encountered during policy verification: %w", err)
+	}
+
+	pass := true
+	for _, result := range results {
+		p := result.Analyze()
+		if !p {
+			pass = false
+		}
 	}
```

Comment:
Can we do this at the end of pol.Verify? it's a bit confusing for the Verify function to return without an error if the policy is actually failing.

Reply:
So I'm glad that this has come up, it's something I wasn't 100% sure on.

The intention here was to change the usage of `err` so it is reserved for functional errors / issues during the `Verify` flow (e.g., `failed to open file`). This is why the `bool` was introduced as a clear way of signifying if the policy verification failed or passed. I was however also concerned that this would end up confusing for users.